### ISSUE TITLE
--fix strips out @ts-ignore comment

### ISSUE THREAD

**Issue by roselpadilla1** [SeinopSys/eslint-plugin-import-newlines#7]
I have a file where I need the `// @ts-ignore` comment above a certain line:

<img width="266" alt="Screen Shot 2022-01-06 at 12 48 42 AM" src="https://user-images.githubusercontent.com/86622739/148335093-1025cd34-9df2-4e6d-b1ed-d594b8a67621.png">

<img width="777" alt="Screen Shot 2022-01-06 at 12 50 07 AM" src="https://user-images.githubusercontent.com/86622739/148335218-c0ab0c4d-dbbc-4ca8-8522-fdcfbeb62843.png">

However, the comment gets striped out once --fix is ran or the editors autoformat feature kicks in.

<img width="325" alt="Screen Shot 2022-01-06 at 12 50 48 AM" src="https://user-images.githubusercontent.com/86622739/148335276-6d2d8fd2-cb45-497c-8fe2-111a50139b41.png">

**Comment by roselpadilla1** [comment]
I was able to get a very crude version to work that allows line comments inside `ImportDeclarations` by doing the following:

```
const lineCommentsInside = node.parent.comments.length > 0  ? 
    node.parent.comments.filter((comment) => comment.type === 'Line' && ( 
        comment.loc.start.line > node.loc.start.line && 
        comment.loc.start.line < node.loc.end.line
    )) : 0;

const numCommentsToInclude = lineCommentsIncluded.length;

...

const expectedLineCount = importedItems + 2 + numCommentsToInclude;
```

Perhaps it's worth adding a new rule `includeLineComments` or `allowComments` that takes into account the comment line count as well? 

I could see someone wanting to: 
- explain *why* they're importing something
- disable some error/warning like no unused imports
- or in my case ts compiler errors/warnings.

**Comment by SeinopSys** [comment]
This is a valid concerns but I cannot provide a timeline for a fix, implementing this adds quite a bit more complexity and it might require substantial refactoring of the rule implementation.

I would strongly suggest that for your specific use case, you create proper typings for the package you're trying to import, by specifying a custom `typeRoots` within your project in `tsconfig.json` ([documented here](https://www.typescriptlang.org/tsconfig#typeRoots)), e.g. `['node_modules/@types', './src/types']`, then create the file `./src/types/react-html5-camera-photo/index.d.ts` with the contents

```typescript
declare module 'react-html5-camera-photo' {
    declare var FACING_MODES: unknown;
    declare var IMAGE_TYPES: unknown;
}
``

You can expand this type further as necessary to include things your project uses and have them type-checked correctly. This is the only immediate solution I can suggest for your specific use case.

**Comment by SeinopSys** [comment]
Actually, I just checked and apparently there's a types package for it, so that may be even easier. Just install `@types/react-html5-camera-photo` in your project and the issue should disappear.

**Comment by roselpadilla1** [comment]
Thank you @SeinopSys you're absolutely right and I have no idea why that package wasn't imported with the other dep. I've gone ahead and addressed that issue locally but others may still encounter similar issues, esp in codebases that are highly annotated (say, for teaching and learning purposes) but this isn't an urgent issue.

Great job on the plugin btw! I liked how it was specifically scoped for import items and not all object-like structures like some others plugins are 👌🏼

**Comment by SeinopSys** [comment]
Since another recently opened issue required the processing of comments I revisited this one as well and released a fix for this specific use case, available as of version 1.3.3

If the problem persists please reopen the issue.
